x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/trusted_apps_list.tsx

```diff
@@ -0,0 +1,125 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useCallback, useMemo } from 'react';
+import { useHistory } from 'react-router-dom';
+import { EuiBasicTable } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { Immutable } from '../../../../../common/endpoint/types';
+import { TrustedApp } from '../../../../../common/endpoint/types/trusted_apps';
+import { MANAGEMENT_PAGE_SIZE_OPTIONS } from '../../../common/constants';
+import { getTrustedAppsListPath } from '../../../common/routing';
+
+import {
+  getListCurrentPageIndex,
+  getListCurrentPageSize,
+  getListErrorMessage,
+  getListItems,
+  getListTotalItemsCount,
+  isListLoading,
+} from '../store/selectors';
+
+import { useTrustedAppsSelector } from './hooks';
+
+import { FormattedDate } from '../../../../common/components/formatted_date';
+
+const OS_TITLES: { [K in TrustedApp['os']]: string } = {
+  windows: i18n.translate('xpack.securitySolution.trustedapps.os.windows', {
+    defaultMessage: 'Windows',
+  }),
+  macos: i18n.translate('xpack.securitySolution.trustedapps.os.macos', {
+    defaultMessage: 'Mac OS',
+  }),
+  linux: i18n.translate('xpack.securitySolution.trustedapps.os.linux', {
+    defaultMessage: 'Linux',
+  }),
+};
+
+const COLUMN_TITLES: { [K in keyof Omit<TrustedApp, 'id' | 'entries'>]: string } = {
+  name: i18n.translate('xpack.securitySolution.trustedapps.list.columns.name', {
+    defaultMessage: 'Name',
+  }),
+  os: i18n.translate('xpack.securitySolution.trustedapps.list.columns.os', {
+    defaultMessage: 'OS',
+  }),
+  created_at: i18n.translate('xpack.securitySolution.trustedapps.list.columns.createdAt', {
+    defaultMessage: 'Date Created',
+  }),
+  created_by: i18n.translate('xpack.securitySolution.trustedapps.list.columns.createdBy', {
+    defaultMessage: 'Created By',
+  }),
+};
+
+const COLUMN_DEFINITIONS = [
+  {
+    field: 'name',
+    name: COLUMN_TITLES.name,
+  },
+  {
+    field: 'os',
+    name: COLUMN_TITLES.os,
+    render(value: TrustedApp['os'], record: Immutable<TrustedApp>) {
+      return OS_TITLES[value];
+    },
+  },
+  {
+    field: 'created_at',
+    name: COLUMN_TITLES.created_at,
+    render(value: TrustedApp['created_at'], record: Immutable<TrustedApp>) {
+      return (
+        <FormattedDate
+          fieldName={COLUMN_TITLES.created_at}
+          value={value}
+          className="eui-textTruncate"
+        />
+      );
+    },
+  },
+  {
+    field: 'created_by',
+    name: COLUMN_TITLES.created_by,
+  },
+];
+
+export const TrustedAppsList = memo(() => {
+  const pageIndex = useTrustedAppsSelector(getListCurrentPageIndex) - 1;
+  const pageSize = useTrustedAppsSelector(getListCurrentPageSize);
+  const totalItemCount = useTrustedAppsSelector(getListTotalItemsCount);
+  const history = useHistory();
+
+  return (
+    <EuiBasicTable
+      columns={COLUMN_DEFINITIONS}
+      items={[...useTrustedAppsSelector(getListItems)]}
```

Comment:
Consider defining all hooks at the top level of the component, such as `useTrustedAppsSelector()`.  This is because React with throw a nasty error if you have a different number of hooks across renders.  We should be OK with this current implementation, but it would likely need to be refactored again later when we introduce something like the Empty State.

Reply:
Hmm, I didn't knew that. I've moved some out and added couple of useMemo, but I still left some inline. I would suggest we do the refactoring at the point when it's really becomes a problem, at this point it just makes the component function longer without a good reason. As for the empty state situation - I did not intent this component to be used to implement the logic, the intent is to implement it one level above (TrustedAppsPage component), so I do not envision having branching in this component at this point of time. Let's refactor when it's really gets to the point when it becomes necessary, but it's a good remark I'll keep this in mind for developing other components. Do you have any links to some articles about this?